src/test/java/UploadReceiptServletTest.java

```diff
@@ -394,43 +393,31 @@ public void doPostThrowsIfUserIsLoggedOut() throws IOException {
   }
 
   @Test
-  public void doPostThrowsIfDateIsInTheFuture() throws IOException, ReceiptAnalysisException {
+  public void doPostUploadsReceiptWithoutTimestampIfParsedDateIsInTheFuture()
```

Comment:
These test names are getting kinda long. We usually like our test names to be in the format: methodBeingTested_situationBeingTested_expectedResult(). Either in this PR or a separate PR, I would refactor the test names.

Example:
transferMoney_deductsFromSource
transferMoney_addsToDestination
transferMoney_insufficientFunds_throwsException
multiply_twoNegativeValues_returnsPositiveValue
multiply_anythingWithNan_returnsNan
inputIsValid_conversionSucceeds